kotlin/src/test/TestMethods.kt

```diff
@@ -202,23 +186,110 @@ class TestMethods {
+    }
+
+    @test
+    fun testRecentSerial() {
+        val limit: Long? = 9
+        runBlocking {
+            val pair = recentSerial(limit)
+            val dashboards = pair.first
+            val looks = pair.second
+            assertNotNull(dashboards)
+            assertNotNull(looks)
+            val l = limit!!.toInt()
+            assertEquals(l, dashboards.count(), "$l Dashboards")
+            assertEquals(l, looks.count(), "$l Looks")
+        }
+    }
+
+    @test fun testRecent() {
+        val limit: Long? = 9
+        val recentDashboards = sdk.ok<Array<Dashboard>>(sdk.search_dashboards(limit = limit, last_viewed_at = "not null", sorts = "last_viewed_at desc"))
+        val dash2 = sdk.ok<Array<Dashboard>>(sdk.search_dashboards(limit = limit, last_viewed_at = "not null", sorts = "last_viewed_at desc"))
+        val recentLooks = sdk.ok<Array<Look>>(sdk.search_looks(limit = limit, last_viewed_at = "not null", sorts = "last_viewed_at desc"))
+        val look2 = sdk.ok<Array<Look>>(sdk.search_looks(limit = limit, last_viewed_at = "not null", sorts = "last_viewed_at desc"))
+        assertNotNull(recentDashboards)
+        assertNotNull(dash2)
+        assertNotNull(recentLooks)
+        assertNotNull(look2)
+        val l = limit!!.toInt()
+        assertEquals(l, recentDashboards.count(), "$l Dashboards")
+        assertEquals(l, dash2.count(),"$l Dashboards")
+        assertEquals(l, recentLooks.count(), "$l Looks")
+        assertEquals(l, look2.count(), "$l Looks")
+    }
+
+    @ExperimentalUnsignedTypes
+    fun mimeType(data: UByteArray) : String {
+
+        val b = data.get(0)
+        val n = b.toUInt().toInt()
+        return when(n)  {
+            0xFF -> "image/jpeg"
+            0x89 -> "image/png"
+            0x47 -> "image/gif"
+            0x4D, 0x49 -> "image/tiff"
+            0x25 -> "application/pdf"
+            0xD0 -> "application/vnd"
+            0x46 -> "text/plain"
+            else -> "application/octet-stream"
+        }
+    }
+
+    @test fun testThumbnailDownload() {
+        val dashboards = sdk.ok<Array<DashboardBase>>(sdk.all_dashboards("id"))
+        dashboards.forEach { d ->
+            d.id?.let { id ->
+                val svg = sdk.ok<String>(sdk.vector_thumbnail("dashboard", id))
+                assertTrue(svg.contains("<svg"), "Dashboard ${id} should have '<svg'")
+            }
+        }
+    }
+
+    @test fun testImageDownload() {
+        val body = simpleQuery()
+        val query = sdk.ok<Query>(sdk.create_query(body))
+        query.id?.let { id ->
+            // sanity check to make sure the query is valid first
+            val sql = sdk.ok<String>(sdk.run_query(id, "sql"))
+            assertNotNull(sql)
+            assertTrue(sql.contains("SELECT"), "Select statement returned")
+            val png = sdk.ok<ByteArray>(sdk.stream.run_query(id, "png")).toUByteArray()
+            assertNotNull(png)
+            assertEquals(mimeType(png), "image/png", "png is png?")
```

Comment:
Probably audit all of these, but this is backwards.

`assertEquals(expected, actual, message)`

Reply:
Yeah, thanks for the reminder. I'll fix all of those.